packages/server-functions-vite-plugin/src/index.ts

```diff
@@ -0,0 +1,246 @@
+import { fileURLToPath, pathToFileURL } from 'node:url'
+
+import { mkdirSync, readFileSync, writeFileSync } from 'node:fs'
```

Comment:
Can we take inspiration from Astro here and propagate the system-reliant node modules like "fs" and "path" from the root config? https://github.com/withastro/astro/blob/98f9e833a5610cbd6f966a827451318435d31483/packages/astro/src/core/config/config.ts#L95

We've had issues opened in the past where the use of a certain system-specific node module for items like "path" and "fs" have broken things like the reading of the client-manifest.
\+ for the future (if we want to), it would let us customize this or use a dependency injection strategy based on the user's system should it be required.

---

Also, just a preference thing on my part, perhaps, we can just rely on synchronous calls? instead of mixing between sync and async. For me, the blocking nature of sync calls makes more sense for these plugins, since they often dependant on a previous plugin/step and can be the decider for the plugin after the current one.